### PR TITLE
Fix git result lesson

### PR DESCRIPTION
**Pull request by NghiaNTT**

### CODE REVIEW

File: app/models/lesson.rb [hunk 1]
```diff
@@ -3,18 +3,24 @@ class Lesson < ActiveRecord::Base
...
     reject_if: lambda {|attribute| attribute[:word_id].blank?}, allow_destroy: true
 
   before_create :random_words
+  before_update :change_status
 
   private
     def random_words
-    self.words = if category.words.size >= lesson_word_size
-      category.words.limit lesson_word_size
+    self.words = if category.words.size >= Settings.lesson_word_size
```
Comment (huongnguyenmta) [inline app/models/lesson.rb]: indent

File: app/models/lesson.rb [hunk 2]
```diff
@@ -3,18 +3,24 @@ class Lesson < ActiveRecord::Base
...
+      category.words.limit Settings.lesson_word_size
     else
       category.words
     end
+
+     def change_status
+      self.status = self.status.nil?
```
Comment (huongnguyenmta) [inline app/models/lesson.rb]: indent

**Comment by huongnguyenmta** [comment]
OK

File: app/controllers/lessons_controller.rb [hunk 3]
```diff
@@ -24,6 +23,16 @@ def create
     end
   end
 
+  def update
+    if @lesson.update_attributes lesson_params
+      redirect_to @lesson
+      flash[:success] = "save"
```
Comment (dieunb) [inline app/controllers/lessons_controller.rb]: i18n

File: app/controllers/lessons_controller.rb [hunk 4]
```diff
@@ -24,6 +23,16 @@ def create
...
+    else
+      flash[:danger] = "fail"
```
Comment (dieunb) [inline app/controllers/lessons_controller.rb]: i18n

File: app/models/lesson.rb [hunk 5]
```diff
@@ -3,18 +3,24 @@ class Lesson < ActiveRecord::Base
...
   private
     def random_words
-    self.words = if category.words.size >= lesson_word_size
-      category.words.limit lesson_word_size
-    else
-      category.words
+      self.words = if category.words.size >= Settings.lesson_word_size
+        category.words.limit Settings.lesson_word_size
+      else
+        category.words
```
Comment (dieunb) [inline app/models/lesson.rb]: thiếu `end`

**Pull request merged**